### PR TITLE
types: migrate test-infra to testify for enum_test.go

### LINKED ISSUE #28000: migrate test-infra to testify for `enum_test.go`

### PR DESCRIPTION
**Pull request by myccccccc**
### What problem does this PR solve?

Issue Number: close #28000 

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
Thanks for your contribution @myccccccc ! It is neat >_<

**Review by mmyj — APPROVED** [review]
lgtm

File: types/enum_serial_test.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+			{[]string{"a"}, "b", 0},
+			{[]string{"a"}, "1", 1},
+		}
+
+		for _, collation := range []string{mysql.DefaultCollationName, "utf8_unicode_ci"} {
+			for _, test := range tests {
+				e, err := ParseEnum(test.Elems, test.Name, collation)
+				if test.Expected == 0 {
+					require.Error(t, err)
+					require.Equal(t, e.ToNumber(), float64(0))
```
Comment (mmyj) [inline types/enum_serial_test.go:45]: Expected is the second arg.

**Review by mmyj — CHANGES_REQUESTED** [review]

**Review by mmyj — APPROVED** [review]

**Comment by mmyj** [comment]
PTAL @tisonkun, we need one more lgtm.

**Review by tisonkun — APPROVED** [review]
Thanks for your review @mmyj !

**Pull request merged**